### ISSUE TITLE
Number of particles with add_process

### ISSUE THREAD

**Issue by roiser** [madgraph5/madgraph4gpu#667]
I am creating a process

```
generate p p > t t~
add process p p > t t~ j
add process p p > t t~ j j
```

which produces code which would not compile. Potentially the problem stems from the fact that the different processes have various numbers of particles in the final state (2, 3 and 4)

I believe the main problem is the variable  ```const int nparf``` in ```src/mgOnGpuConfig.h``` which is set to the same number for all processes (2 in that case). 

The compilation error observed then shows up in "a" CPPProcess.cc e.g. for variable ```static constexpr short tHel[ncomb][mgOnGpu::npar]``` 

Please note that I believe also the ncomb actually is wrong. We should double check that this is also set correctly.

**Comment by valassi** [comment]
Thanks Stefan I am starting to prepare a patch

**Comment by valassi** [comment]
Ok the patch is essentially ready. I am running tests, will see tomorrow. This is in #668

**Comment by valassi** [comment]
I have rerun tests, but there are still somepeculiar issues I want to have a quick look at
- Performance degrades slightly, with a higher number of GPU registers?!
- Tmad tests fail with cross section mismatches? (this may be an issue on the node I am using)

**Comment by valassi** [comment]
About the two issues above:
- I fixed the tmad bug causing errors on itscrd90 (python vs python3)
- I opened #669 and understood the performance regression: the cause is the simultanenous move from int to size_t, which I will undo

**Comment by valassi** [comment]
Closed by merging #668
